.github/workflows/nightly-test.yml

```diff
@@ -0,0 +1,17 @@
+name: Gradle Nightly Test
+on:
+  schedule:
+    - cron: 0 0 * * * # daily
+jobs:
+  nightly-test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-java@v3
+      with:
+        distribution: temurin
+        java-version: 17
+    - uses: gradle/gradle-build-action@v2
+      with:
+        gradle-version: nightly
+    - run: gradle test # just test build configuration
```

Comment:
This will take many many many hours, this just isnt how the test are ran on GHA. Take a look at the other workflow, ideally this cron task could be added to the other one.